dv/cosim/spike_cosim.cc

```diff
@@ -585,11 +633,15 @@ void SpikeCosim::set_nmi(bool nmi) {
     mstack.mpie = get_field(processor->get_csr(CSR_MSTATUS), MSTATUS_MPIE);
     mstack.epc = processor->get_csr(CSR_MEPC);
     mstack.cause = processor->get_csr(CSR_MCAUSE);
+
+    early_interrupt_handle();
   }
 }
 
 void SpikeCosim::set_nmi_int(bool nmi_int) {
-  if (nmi_int && !nmi_mode && !processor->get_state()->debug_mode) {
+  if (nmi_int && !nmi_mode && !processor->get_state()->debug_mode &&
+      processor->halt_request != processor_t::HR_REGULAR) {
+    bool old_nmi_int = processor->get_state()->nmi_int;
```

Comment:
This value doesn't seem to be used. Do we need this line?

Reply:
Ah whoops we don't, failed to delete it after making some changes.